src/accelerator/ogl/image/image_kernel.cpp

```diff
@@ -234,9 +234,66 @@ struct image_kernel::impl
             params.layer_key->bind(static_cast<int>(texture_id::layer_key));
         }
 
-        // Setup shader
+        // Setup conversion
 
         shader_->use();
+        std::shared_ptr<texture> conversion_texture;
+
+        if (params.pix_desc.format == core::pixel_format::uyvy) {
+            int width  = params.pix_desc.planes.at(0).width;
```

Comment:
Does this need to be a separate call into the shader?
The normal shader call should be able to handle doing the colour conversion as part of the mix. Which is how the planar yuv that ffmpeg produces is handled currently

Reply:
You can convert anything planar, semi-planar like NV12, or packed 4:4:4(:4) while mixing, but can you do it when components are interleaved like in UYVY? I couldn't find a way to make linear interpolation work correctly with it, hence the decision to first draw the frame in 1:1 scale.

Reply:
Good point, I forgot about that